Dockerfile

```
@@ -0,0 +1,33 @@
+RUN apk add --no-cache ca-certificates
+
+# Download dependencies first for better layer caching
+COPY go.mod go.sum ./
+RUN go mod download
+
+# Copy source and build
+COPY . .
+RUN CGO_ENABLED=0 GOOS=linux go build -ldflags="-s -w" -o /gomodel ./cmd/gomodel
+
+# Runtime stage
+FROM gcr.io/distroless/static-debian12
```

Hadolint (2.14.0)

[warning] Always tag the version of an image explicitly

(DL3006)